checker.c

```diff
@@ -1,21 +1,85 @@
+/* Author : Abhijeet C */
+/* Program : Primitive BMS simulation */
+
+/* Includes */
 #include <stdio.h>
 #include <assert.h>
+#include <stdlib.h>
+#include <math.h>
+#include <stdbool.h>
+
+/* defines */
+
+#define MAXCHARGERATE 0.8
+#define MAXTEMP 45
+#define MINTEMP 0
+#define MAXSOC 80
+#define MINSOC 20
+
+
+/* Check for temperature range */
+/* Arguement : Temperature  */
+int checktemperature(float TempIn)
```

Comment:
The check... functions are very similiar. Reduce duplication by isolating the range-check into a function and call it with different ranges.